pkg/image/api/types.go

```diff
@@ -60,6 +77,41 @@ type ImageLayer struct {
 	Size int64
 }
 
+// SignatureValidationResult holds a result of signature validation.
+type SignatureValidationResult string
+
+// ImageSignature holds a signature of an image.
+type ImageSignature struct {
+	// Describes a type of stored blob. Provided by caller.
+	Type string
+	// An opaque binary string which is an image's signature.
+	Content []byte
+	// A time of last server-side validation.
+	ValidatedOn unversioned.Time
+	// Result of validation performed at ValidatedOn.
+	ValidationResult SignatureValidationResult
+
+	// At most one of the following pointers can point to details. If the signature type isn't recognized,
+	// they will all be nil.
+
+	// Details for signature of type "atomic".
+	AtomicSignatureDetails *AtomicSignatureDetails
+	// For any future supported signature type, add additional pointer to its own details struct.
+}
+
+// AtomicSignatureDetails holds metadata of recognized signature of type "atomic". They are parsed from
+// signature's content.
+type AtomicSignatureDetails struct {
+	// Digest of signed manifest.
+	SignedImageName string
+	// Docker image reference
+	SignedDockerImageReference string
```

Comment:
> These references are almost never going to match real values in a
production system.  For one thing, we want images to be pullable from
multiple sources.

How the image has been copied into the system is AFAICT more or less irrelevant to its identity.

> It's useful to attest to the name, but the digest is the
value that we should be verifying against.

We are _also_ verifying the digest (the `signedImageName` field), but that is insufficient for handling the “horribly obsolete RHEL5 image that we definitely don’t want to use but which also needs to be signed by Red Hat” case. 

Yes, the policy evaluated by OpenShift will need to know about the mappings between sources, imagestreams, and expected identities. I can’t see an alternative, though. Am I missing something obvious?